modules/exploits/windows/ftp/quickshare_traversal_write.rb

```diff
@@ -147,11 +146,16 @@ def exploit
 
     begin
       t = framework.threads.spawn("reqs", false) {
-        # Upload our malicious executable
-        u = upload(exe_name)
-
-        # Upload the mof file
-        upload(mof_name) if u
+        begin
+          # Upload our malicious executable
+          u = upload(exe_name)
+
+          # Upload the mof file
+          upload(mof_name) if u
+        rescue ::Exception => e
+          print_error e.message
+          cleanup
```

Comment:
I haven't looked too deep into this code - but just just confirming, should this be `ensure cleanup` ? 👀

Reply:
Ah, I'm guessing super does something with the uploaded file - don't mind me :+1:

Reply:
You are correct, `cleanup` would normally be called later by `super`.  In this case when the thread fails and exits,`super` is actually what hangs indefinitely so we only want to cleanup if this block errors.